src/main/java/net/sf/jabref/model/entry/BibEntry.java

```diff
@@ -611,4 +612,21 @@ public void addKeywords(List<String> keywords) {
         // TODO Auto-generated method stub
         return fields;
     }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) {
+            return true;
+        }
+        if (o == null || getClass() != o.getClass()) {
+            return false;
+        }
+        BibEntry entry = (BibEntry) o;
+        return Objects.equals(type, entry.type) && Objects.equals(fields, entry.fields);
```

Comment:
What about the comments before the entry? I thought, these are also stored in the entry, aren't they?